src/Framework/TestCase.php

```diff
@@ -1724,31 +1752,54 @@ private function handleDependencies(): bool
                     $dependency = $className . '::' . $dependency;
                 }
 
+                if (\strpos($dependency, '\\') === 0) {
+                    $dependency = \substr($dependency, 1);
+                }
+
                 if (!isset($passedKeys[$dependency])) {
-                    $this->status = BaseTestRunner::STATUS_SKIPPED;
-
-                    $this->result->startTest($this);
-
-                    $this->result->addError(
-                        $this,
-                        new SkippedTestError(
-                            \sprintf(
-                                'This test depends on "%s" to pass.',
-                                $dependency
-                            )
-                        ),
-                        0
-                    );
+                    if (
+                        !isset($skippedKeys[$dependency])
+                        && !isset($failedKeys[$dependency])
+                        && \strpos($dependency, '::') > 0
+                        && $className !== \explode('::', $dependency, 2)[0]
+                    ) {
+                        $dependencyClass = \explode('::', $dependency, 2)[0];
+                        (new TestSuite($dependencyClass, 'Dependency_' . \ucfirst($dependencyClass)))
+                            ->run($this->result);
```

Comment:
😨 This is scary bit of code. It fires off a lot of logic that will have unforeseen consequences, I'm just not sure which at the moment.

If I am reading this correctly, this code does the following:
1. if the `$dependency` has not been passed yet, do
2. check if `$dependency` hasn't been failed or skipped
3. get the name of the class in which the dependency resides by splitting the FQCN
4. yolo `(new TestSuite($theDependencyWeNeed))->run($currentResult)`

I get what you are trying to achieve: auto-run dependencies without any hassle on the user side. If this logic were to be accepted, it needs more safeguards en refactoring:
- what are the side effects of instantiating a `TestSuite` here in this scope?
- what happens if this `TestSuite` has already been instantiated, but: not run? or not run completely? failed during either `setUpBeforeClass` or `setUp`?
- even if this works, `handleDependencies()` shouldn't just `->run()` the missing dependency and merge it into the current result. Leave that responsibility to the Runner and find a cleaner way to pass around the execution plan.

Tests, tests and more tests. Preferably a few end-to-end ones :-)

Reply:
> This is scary bit of code. It fires off a lot of logic
I know... Kinda the way to deal with it, when I cannot access the list of tests to execute. Another solution would be scanning all files which depends on which one and reorder them. However, I tried it without success.

> If I am reading this correctly, this code does the following:
Yup.

> what are the side effects of instantiating a TestSuite here in this scope?
![image](https://user-images.githubusercontent.com/4658966/49815552-b5a01100-fd74-11e8-97ee-4e2894a76fa7.png)

> what happens if this TestSuite has already been instantiated, but: not run? or not run completely? failed during either setUpBeforeClass or setUp
Good question. I will check what happens.

> Tests, tests and more tests. Preferably a few end-to-end ones :-)
Not sure how to implement those at all...

> even if this works, handleDependencies() shouldn't just ->run() the missing dependency and merge it into the current result. Leave that responsibility to the Runner and find a cleaner way to pass around the execution plan.
More specific implementation how to do it?
I could just run it like this and remove the result from wrong place like shown in the image?

Reply:
FYI, this issue showing in the image has been fixed..